lib/common/components/FormInput.js

```diff
@@ -0,0 +1,70 @@
+// @flow
+
+import React from 'react'
+import {
+  Col,
+  FormGroup,
+  ControlLabel,
+  FormControl
+} from 'react-bootstrap'
+
+import {getComponentMessages} from '../../common/util/config'
+import type {FormProps} from '../../types'
+
+type Props = {
+  field: FormProps,
+  messageRoot?: string,
+  onChange: any,
+  shouldRender: () => boolean,
+  value?: any
+}
+
+export default class FormInput extends React.Component<Props> {
+  static defaultProps = {
+    shouldRender: () => true
+  }
+
+  messages = getComponentMessages('FormInput')
+
+  render () {
+    const {field, messageRoot, shouldRender, value} = this.props
+    const prefix = messageRoot ? `${messageRoot}.` : ''
+    const {effects, split, ...fieldProps} = field
+    // If should render is false, do not render form element (for conditional
+    // visibility of form elements).
+    if (!shouldRender()) return null
+    const isCheckbox = field.type === 'checkbox'
+    const isSelect = field.type.indexOf('select') !== -1
+    const checkboxProps = isCheckbox ? {checked: value} : {}
```

Comment:
Definitely looking like a new form framework. Gonna pitch [React Formal](http://jquense.github.io/react-formal/) again.

Reply:
For this PR, I don't think it's a good idea to replace the form framework entirely. I'm just working within the confines of what is already set up (I did do some refactoring/simplification of the form components). But I do agree that it might be a good call to replace this soon.